File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++)
        {
            mp[nums[i]]=i;
        }

        for(int i=0;i<nums.size();i++)
        {
            int temp=target-nums[i];
            if(mp.count(temp) && mp[temp]!=i)
            {
                return {i,mp[temp]};
            }
        }
        return {};
        
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp={}

        for i , num in enumerate(nums):
            k=target-num
            if k in temp:
                return [temp[k],i]
            temp[num]=i
        return []
        